docs_src/getting-started/Ch-GettingStartedSnapUsers.md

```diff
@@ -6,170 +6,733 @@
+    `apps.secrets-config.proxy.admin.public-key`.
+
+    This is particularly useful when seeding the snap from a [Gadget](https://snapcraft.io/docs/gadget-snap) on an [Ubuntu Core](https://ubuntu.com/core) system.
+
+!!! example "Generating a JWT token for the example user"
+    On success, a JWT token is printed out and written to user.jwt file.
+    We use the user id `1000` as set in the previous example.
+    ```bash
+    edgexfoundry.secrets-config proxy jwt --algorithm ES256 --private_key private.pem --id 1000 --expiration=1h | tee user.jwt
+    ```
+
+It is also possible to create the JWT token using bash and openssl. But that is beyond the scope of this guide.
+
+Once you have the token, you can access the services via the API Gateway.
+
+!!! example "Calling an API on behalf of example user"
+
+    ```bash
+    curl --insecure https://localhost:8443/core-data/api/v2/ping -H "Authorization: Bearer $(cat user.jwt)"
+    ```
+    Output: `{"apiVersion":"v2","timestamp":"Mon May  2 12:14:17 CEST 2022","serviceName":"core-data"}`
+
+#### Accessing Consul
+Consul API and UI can be accessed using the consul token (Secret ID). For the snap, token is the value of `SecretID` typically placed in a JSON file at `/var/snap/edgexfoundry/current/secrets/consul-acl-token/bootstrap_token.json`.
+
+!!! example
+    To get the token:
+    ```bash
+    sudo cat /var/snap/edgexfoundry/current/secrets/consul-acl-token/bootstrap_token.json | jq '.SecretID'
+    ```
+    Output: `"ee3964d0-505f-6b62-4c88-0d29a8226daa"`
+
+    Try it out locally:
+    ```bash
+    curl --insecure --silent http://localhost:8500/v1/kv/edgex/core/2.0/core-data/Service/Port -H "X-Consul-Token:17938174-059e-8b86-122a-9a08a99dcd1c"
+    ```
+
+    Through the API Gateway:  
+    We need to pass both the Consul token and Secret Store token obtained in [Adding API Gateway users](#adding-api-gateway-users) examples.
+    ```bash
+    curl --insecure --silent https://localhost:8443/consul/v1/kv/edgex/core/2.0/core-data/Service/Port -H "X-Consul-Token:17938174-059e-8b86-122a-9a08a99dcd1c" -H "Authorization: Bearer $(cat user.jwt)"
+    ```
+
+#### Setting TLS certificates
+The API Gateway setup generates a self-signed certificate by default. To replace that with your own certificate, refer to API Gateway guide: [Using a bring-your-own external TLS certificate for API gateway](../../security/Ch-APIGateway/#using-a-bring-your-own-external-tls-certificate-for-api-gateway) and use the snapped `edgexfoundry.secrets-config` utility.
+
+To get the usage help:
+```bash
+edgexfoundry.secrets-config proxy tls -h
+```
+You may also refer to the [secrets-config proxy](../../security/secrets-config-proxy/) documentation.
+
+!!! example 
+    Given certificate `cert.pem`, private key `privkey.pem`, and certificate authority `ca.pem` files:
```

Comment:
how to get `cert.pem`, `privkey.pem`, and `ca.pem`?

Reply:
They may be created by a CA or manually as self-signed certificates. Many ways to create them. We can add self-signed one in future, although it is beyond the scope of a getting started. E.g. for self signed: https://github.com/canonical/edgex-checkbox-provider/blob/c363c0a301fd2e102e828496365f637e00fe3edb/data/utils.sh#L182-L191

Reply:
The section starts with "The API Gateway setup generates a self-signed certificate by default. To replace that **with your own certificate**..."

Reply:
Extended the example to clarify.

Reply:
then the section would be `Setting TLS certificates with your own certificate`, right?